python/lsst/analysis/tools/tasks/base.py

```diff
@@ -246,3 +249,57 @@ def run(self, *, data: KeyedData | None = None, **kwargs) -> Struct:
             setattr(results, name, value)
 
         return results
+
+    def runQuantum(
+        self,
+        butlerQC: ButlerQuantumContext,
+        inputRefs: InputQuantizedConnection,
+        outputRefs: OutputQuantizedConnection,
+    ) -> None:
+        """Override default runQuantum to load the minimal columns necessary
+        to complete the action.
+
+        Parameters
+        ----------
+        butlerQC : `ButlerQuantumContext`
+            A butler which is specialized to operate in the context of a
+            `lsst.daf.butler.Quantum`.
+        inputRefs : `InputQuantizedConnection`
+            Datastructure whose attribute names are the names that identify
+            connections defined in corresponding `PipelineTaskConnections`
+            class. The values of these attributes are the
+            `lsst.daf.butler.DatasetRef` objects associated with the defined
+            input/prerequisite connections.
+        outputRefs : `OutputQuantizedConnection`
+            Datastructure whose attribute names are the names that identify
+            connections defined in corresponding `PipelineTaskConnections`
+            class. The values of these attributes are the
+            `lsst.daf.butler.DatasetRef` objects associated with the defined
+            output connections.
+        """
+        inputs = butlerQC.get(inputRefs)
+        data = self.loadData(inputs["data"])
+        outputs = self.run(data=data)
+        butlerQC.put(outputs, outputRefs)
+
+    def loadData(self, handle: DeferredDatasetHandle):
+        """Load the minimal set of columns from the input dataset."""
```

Comment:
I was thinking about this, and I think perhaps this should be passed the columns instead of getting them itself. Even though it ends up being a 1 line function, I think it is a useful abstraction and provides a hook to be overridden if needed.

I am just thinking it there are more than one inputs that get joined together, not all columns will apply to all tables, and a task might need to split up in some way which handle fetches which column.

Thoughts?

Reply:
Maybe it's better to pass `**kwargs` to `loadData` and `collectInputNames`? Then the default behavior will still work, but will allow an override. Alternatively, just `loadData` could accept `columns` as a parameter that is `None` unless overridden by the user.

Personally I prefer using `kwargs` but not so strongly that I wouldn't mind something else.

Reply:
I like the idea that the signature of `loadData` would be `(self, handle: DeferredDatasetHandle, names: Iterable[str] | None = None) -> KeyedData` Likely you will need to cast to KeyedData unless the butler is returning Any. Then if it is None load it for the user like you do now.